File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    let count = 1;
    for(let i=1;i<nums.length;i++){
        if(nums[i] != nums[i-1]){
            nums[count] = nums[i];
            count++;
        }
    }
    return count;
};

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        int index = 0;
        for(int num: nums){
            if(!set.contains(num)){
                set.add(num);
                nums[index++] = num;
            }
        }
        return index;
    }
}